File: subsets-ii.cpp
class Solution {
    public static void Met(int i,int[]nums,List<List<Integer>>a,List<Integer>ds)
    {
        if(i==nums.length){
            if(!a.contains(new ArrayList<>(ds)))
            {
            a.add(new ArrayList<>(ds));
                
            }
            return;
        }
        ds.add(nums[i]);
        Met(i+1,nums,a,ds);
        ds.remove(ds.size()-1);
        Met(i+1,nums,a,ds);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>>a=new ArrayList<>();
        Met(0,nums,a,new ArrayList<>());
        return a;
    }
}

File: subsets-ii.java
class Solution {
   public void operation(int n,int[] nums,int i,ArrayList<Integer> temp,List<List<Integer>> ans){
        if(i>=n){
            if(!ans.contains(new ArrayList<>(temp)))
            ans.add(new ArrayList<>(temp));
            return;
        }
        temp.add(nums[i]);
        operation(n,nums,i+1,temp,ans);
        temp.remove(temp.size()-1);
        operation(n,nums,i+1,temp,ans);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        ArrayList<Integer> temp = new ArrayList<>();
        List<List<Integer>> ans = new ArrayList<>();
        operation(n,nums,0,temp,ans);
        return ans;
    }
}
